roadmap/implementors-guide/guide.md

```diff
@@ -883,8 +883,67 @@ Included: Option<()>,
+  1. For security, validators that misbehave shouldn't only be slashed on one fork, but on all possible forks. Validators that misbehave shouldn't be able to create a new fork of the chain when caught and get away with their misbehavior.
+  2. It is possible that the parablock being contested has not appeared on all forks.
+  3. If a block author believes that there is a disputed parablock on a specific fork that will resolve to a reversion of the fork, that block author is better incentivized to build on a different fork which does not include that parablock.
+
+This means that in all likelihood, there is the possibility of disputes that are started on one fork of the relay chain, and as soon as the dispute resolution process starts to indicate that the parablock is indeed invalid, that fork of the relay chain will be abandoned and the dispute will never be fully resolved on that chain.
+
+Even if this doesn't happen, there is the possibility that there are two disputes underway, and one resolves leading to a reversion of the chain before the other has concluded. In this case we want to both transplant the concluded dispute onto other forks of the chain as well as the unconcluded dispute.
+
+We account for these requirements by having the validity module handle two kinds of disputes.
+  1. Local disputes: those contesting the validity of the current fork by disputing a parablock included within it.
+  2. Remote disputes: a dispute that has partially or fully resolved on another fork which is transplanted to the local fork for completion and eventual slashing.
+
+#### Local Disputes
+
 [TODO: store all included candidate and attestations on them here. accept additional backing after the fact. accept reports based on VRF. candidate included in session S should only be reported on by validator keys from session S. trigger slashing. probably only slash for session S even if the report was submitted in session S+k because it is hard to unify identity]
 
+One first question is to ask why different logic for local disputes is necessary. It seems that local disputes are necessary in order to create the first escalation that leads to block producers abandoning the chain and making remote disputes possible.
+
+Local disputes are only allowed on parablocks that have been included on the local chain and are in the acceptance period.
+
+For each such parablock, it is guaranteed by the inclusion pipeline that the parablock is available and the relevant validation code is available.
+
+Disputes may occur against blocks that have happened in the session prior to the current one, from the perspective of the chain. In this case, the prior validator set is responsible for handling the dispute and to do so with their keys from the last session. This means that validator duty actually extends 1 session beyond leaving the validator set.
```

Comment:
Yes, the VRF keys used should be whatever keys existed when the candidate became available.

Reply:
There is no mention of VRFs in this paragraph. But if you are referring to the remote disputes section, then I actually disagree with you, because there may be multiple forks where the candidate became available, so there is no clear definition of your statement.